library/include/rocwmma/internal/dpp_impl.hpp

```diff
@@ -103,13 +104,20 @@ namespace rocwmma
                           typename DataT>
                 ROCWMMA_DEVICE static inline DataT exec(DataT src0, DataT src1)
                 {
-                    reinterpret_cast<int32_t&>(src0) = __builtin_amdgcn_update_dpp(
-                        reinterpret_cast<int32_t const&>(src1), // fill value 'prev'
-                        reinterpret_cast<int32_t const&>(src0), // Src value
-                        DppCtrl::opCtrl(), // DPP control code
-                        WriteRowMask, // Mask for affected rows
-                        WriteBankMask, // Mask for affected banks
-                        BoundCtrl); // Fill in 0 on invalid indices
+                    constexpr size_t VecSize = sizeof(DataT) / sizeof(uint32_t);
+                    auto             dpp     = [](auto&& idx, auto&& v0, auto&& v1) {
+                        constexpr auto i = std::decay_t<decltype(idx)>::value;
+                        *(reinterpret_cast<uint32_t*>(&v0) + i) = __builtin_amdgcn_update_dpp(
```

Comment:
To make this easier to read, we might try to cast the input vector to uint32_t[VecSize] and index via i

Reply:
changed